go/pkg/client/cas.go

```diff
@@ -940,12 +969,13 @@ func (c *Client) readBlobStreamed(ctx context.Context, hash string, sizeBytes, o
 }
 
 // ReadProto reads a blob from the CAS and unmarshals it into the given message.
-func (c *Client) ReadProto(ctx context.Context, d digest.Digest, msg proto.Message) error {
-	bytes, err := c.ReadBlob(ctx, d)
+// Returns the size of the proto and the amount of bytes moved through the wire.
+func (c *Client) ReadProto(ctx context.Context, d digest.Digest, msg proto.Message) (int64, int64, error) {
```

Comment:
nit: maybe return the metadata everywhere instead of one of two ints? That way, we never have to change the API again.

Reply:
I think that would be convenient, but `Requested` and `Cached` are higher level concepts than "real bytes" and "compressed bytes", and don't necessarily make much sense at this level. Wdyt?

Reply:
Idk, I'm only willing to be that we will get invariably confused at which functions return which numbers... Also, I'm particularly worried about changing public function APIs unnecessarily.
The only reason Cached doesn't apply here is because it is only dealt with in the *IfMissing flow, right? But everything else should apply, no?
Anyway, up to you, I'm just worried that it's confusing and will still change a bunch...

Reply:
I changed it to your suggestion and tbh it does look way nicer now :)